src/lib/config/config-validator.go

```diff
@@ -1,61 +1,131 @@
+				intervalErr = append(intervalErr, errors.New("`test` property for interval #"+strconv.Itoa(i+1)+" is missing"))
 			}
-			if interval.Type == "" {
-				log.Errorln("`type` property for interval #" + strconv.Itoa(i+1) + " is missing.")
+			if !durationType[interval.Type] {
+				intervalErr = append(intervalErr, errors.New("`type` property for interval #"+strconv.Itoa(i+1)+" is not valid, It must be sec, min or hr for second, minute and hour respectively"))
 			}
 			if interval.Duration == nil {
-				log.Errorln("`duration` property for interval #" + strconv.Itoa(i+1) + " is missing.")
+				intervalErr = append(intervalErr, errors.New("`duration` property for interval #"+strconv.Itoa(i+1)+" is missing"))
 			}
 		}
 	}
+	return intervalErr
 }
 
 // ValidateUtilsConf validates the `utils` property
 // in the configuration file.
-func ValidateUtilsConf(config *UConfig) {
+func ValidateUtilsConf(config *UConfig) []error {
+	utilsErr := []error{}
 	if config.ServicesSignal.FloodPing == "" {
-		log.Errorln("`flood-ping` property under `Services-state` is missing.")
+		utilsErr = append(utilsErr, errors.New("`flood-ping` property under `Services-state` is missing"))
 	}
 	if config.ServicesSignal.Jitter == "" {
-		log.Errorln("`jitter` property under `Services-state` is missing.")
+		utilsErr = append(utilsErr, errors.New("`jitter` property under `Services-state` is missing"))
 	}
 	if config.ServicesSignal.Ping == "" {
-		log.Errorln("`ping` property under `Services-state` is missing.")
+		utilsErr = append(utilsErr, errors.New("`ping` property under `Services-state` is missing"))
 	}
 	if config.ServicesSignal.ReqResDelayMonitoring == "" {
-		log.Errorln("`req-res-delay-or-monitoring` property under `Services-state` is missing.")
+		utilsErr = append(utilsErr, errors.New("`req-res-delay-or-monitoring` property under `Services-state` is missing"))
 	}
+	return utilsErr
 }
 
 // ValidatePasswordProp validates the `password` property in
 // the configuration file.
-func ValidatePasswordProp(password string) {
+func ValidatePasswordProp(password string) error {
 	if password == "" {
-		log.Errorln("`password` property is missing.")
+		return errors.New("password should not be empty")
 	}
+	return nil
+}
+
+func ValidateRoutes(routes *[]Route) []error {
+	Method := map[string]bool{
+		"POST":    true,
+		"GET":     true,
+		"DELETE":  true,
+		"PUT":     true,
+		"PATCH":   true,
+		"COPY":    true,
+		"HEAD":    true,
+		"OPTIONS": true,
+	}
+	routesErr := []error{}
+	for _, route := range *routes {
+		if !valid.IsURL(route.URL) {
+			routesErr = append(routesErr, fmt.Errorf("\"%s\" is not a valid URL, Update the URL", route.URL))
+		}
+		if !Method[route.Method] {
+			routesErr = append(routesErr, fmt.Errorf("for \"%s\" \"%s\" is not a valid http method, Update the Method", route.URL, route.Method))
+		}
+	}
+	return routesErr
 }
 
 // Validate validates the local configuration file.
 func (inst *Config) Validate() bool {
```

Comment:
I think we can return early here (if logging everything is not so important to us). If the first test fails we do not have to wait for other tests appending into the `errors` variable. Let's return a true at the end so that this function returns `isValid` only when all tests pass. Let me know what you think.

Reply:
My thinking over here was. If user gets his first error he will fix it and again run the server.
At step two he will find second error again he will follow similar approach. And so on.
So printing all the errors in the config file at once will help him fix all the error instantly. Without making him run the server multiple times.

Reply:
Fair point! Let's keep it as it is then.

Reply:
Thanks🙃